### ISSUE TITLE
Error building for production

### ISSUE THREAD

**Issue by supardyke** [MisterJimson/flutter_keyboard_visibility#51]
/Users/hdd/.pub-cache/hosted/pub.dartlang.org/flutter_keyboard_visibility-4.0.2/android/src/main/java/com/jrai/flutter_keyboard_visibility/FlutterKeyboardVisibilityPlugin.java
:9: error: cannot find symbol
           import androidx.annotation.NonNull;
                                     ^
             symbol:   class NonNull
             location: package androidx.annotation

**Comment by MisterJimson** [comment]
You may need to properly enable Android X for your app: https://flutter.dev/docs/development/androidx-migration

**Comment by rodrigolmacedo** [comment]
same error here, and my project already is android X

**Comment by MisterJimson** [comment]
So this is definitely related to project or local environment setup, but I may able to help.

I've removed the Android X import and annotations that normally are present when you create a plugin in Java. Can I get some help testing it as I cannot reproduce the issue?

Test your app in both release and debug mode with this version: `4.0.3-beta.0`

**Comment by MisterJimson** [comment]
If you have had a chance to check that version I would love feedback!

**Comment by rodrigolmacedo** [comment]
> 
> 
> If you have had a chance to check that version I would love feedback!

now works fine here, but idk what solved the problem, im using that version 4.0.3-beta.0 and upgraded version of gradle.(4.0.1 with 6.5-all)

**Comment by ffactory-ofcl** [comment]
Yay thanks! 🎉 

If anyone has the same issue (as I had):
I **solved** it by doing these steps:
- Upgrade pubspec.yaml from `flutter_keyboard_visibility: ^4.0.2` to `flutter_keyboard_visibility: ^4.0.3-beta.0`
- Upgrade android/build.gradle from `classpath 'com.android.tools.build:gradle:3.5.0'` to `classpath 'com.android.tools.build:gradle:4.1.0'`
- Remove in gradle.properties this line: `android.enableR8=false`
- [Not sure if this step is related] Replace in android/app/src/main/kotlin/ml/fourinarow_app/MainActivity.kt
this

    ```
    package ml.fourinarow_app

    import io.flutter.embedding.android.FlutterActivity

        class MainActivity: FlutterActivity() {
    }
    ```
    with this
    ```
    package ml.fourinarow

    import androidx.annotation.NonNull;
    import io.flutter.embedding.android.FlutterActivity
    import io.flutter.embedding.engine.FlutterEngine
    import io.flutter.plugins.GeneratedPluginRegistrant

    class MainActivity: FlutterActivity() {
        override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
            GeneratedPluginRegistrant.registerWith(flutterEngine);
        }
    }
    ```
- Upgrade android/gradle/wrapper/gradle-wrapper.properties from ```distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip``` to `distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip`

**Comment by MisterJimson** [comment]
I released `4.0.3`  and `5.0.0-nullsafety.1` with the change from `^4.0.3-beta.0`

**Comment by ollydixon** [comment]
@MisterJimson seeing this issue again with ```flutter_keyboard_visibility: ^5.0.2```
